code/modules/mob/living/living_movement.dm

```diff
@@ -42,12 +42,18 @@
 		remove_movespeed_modifier(MOVESPEED_ID_LIVING_TURF_SPEEDMOD)
 
 /mob/living/proc/update_pull_movespeed()
-	if(pulling && isliving(pulling))
-		var/mob/living/L = pulling
-		if(drag_slowdown && !(L.mobility_flags & MOBILITY_STAND) && !L.buckled && grab_state < GRAB_AGGRESSIVE)
-			add_movespeed_modifier(MOVESPEED_ID_PRONE_DRAGGING, multiplicative_slowdown = PULL_PRONE_SLOWDOWN)
-			return
-	remove_movespeed_modifier(MOVESPEED_ID_PRONE_DRAGGING)
+	if(pulling && (isliving(pulling) || isstructure(pulling)))
+		if(isliving(pulling))
+			var/mob/living/L = pulling
+			if(L.drag_slowdown && !(L.mobility_flags & MOBILITY_STAND) && !L.buckled && grab_state < GRAB_AGGRESSIVE)
+				add_movespeed_modifier(MOVESPEED_ID_BULKY_DRAGGING, multiplicative_slowdown = PULL_BULKY_SLOWDOWN)
+				return
+		if(isstructure(pulling))
+			var/obj/structure/S = pulling
+			if(S.drag_slowdown)
+				add_movespeed_modifier(MOVESPEED_ID_BULKY_DRAGGING, multiplicative_slowdown = PULL_BULKY_SLOWDOWN)
+				return
+	remove_movespeed_modifier(MOVESPEED_ID_BULKY_DRAGGING)

```

Comment:
This entire proc has bad structure. Try again.

Reply:
better?

Reply:
yes